io/src/write_buffer.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2023 The MobileCoin Foundation
```

Comment:
The `std` solution would be to use [`Cursor`](https://doc.rust-lang.org/std/io/struct.Cursor.html) but that felt a bit heavy to bring in here. 
I'm not opposed to it since this is a "made" up interface not in the rust `std`